core/src/test/scala/integration/kafka/tools/MirrorMakerIntegrationTest.scala

```
@@ -0,0 +1,68 @@
+    val brokerList = TestUtils.getBrokerListStrFromServers(servers)
+
+    // Create a test producer to delivery a message
+    val producerProps = new Properties()
+    producerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList)
+    producerProps.put("producer.type", "sync")
+    producerProps.setProperty(ProducerConfig.KEY_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArraySerializer")
+    producerProps.setProperty(ProducerConfig.VALUE_SERIALIZER_CLASS_CONFIG, "org.apache.kafka.common.serialization.ByteArraySerializer")
+    val producer = new MirrorMakerProducer(producerProps)
+    MirrorMaker.producer = producer
+    MirrorMaker.producer.send(new ProducerRecord(topic, msg.getBytes()))
+    MirrorMaker.producer.close()
+
+    servers foreach { server =>
+      println(server.zkUtils.getAllTopics().mkString(","))
+    }
+    // Create a MirrorMaker consumer
+    val consumerProps = new Properties()
+    consumerProps.put(ConsumerConfig.GROUP_ID_CONFIG, "test-group")
+    consumerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList)
+    consumerProps.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "earliest")
+    val consumer = new KafkaConsumer(consumerProps, new ByteArrayDeserializer(), new ByteArrayDeserializer())
+
+    val whitelist = Some("new.*,another_topic,foo")
+    val mirrorMakerConsumer = new MirrorMakerNewConsumer(consumer, None, whitelist)
+    mirrorMakerConsumer.init()
+    try {
+      val maxTryCount = 3 // it might need to call multiple poll calls to retrieve the message
+      for (_ <- 0 until maxTryCount) {
+        try {
+          val data = mirrorMakerConsumer.receive()
+          assertTrue(s"MirrorMaker consumer should get the correct topic: $topic", data.topic == topic)
+          assertTrue("MirrorMaker consumer should read the correct message.", new String(data.value) == msg)
+          return
+        } catch {
+         case _: ConsumerTimeoutException => // swallow it
```

Hmm, this doesn't seem right. Why do we swallow this exception?

---

`val mirrorMakerConsumer = new MirrorMakerNewConsumer(consumer, None, whitelist)

mirrorMakerConsumer.init()

 try {
      TestUtils.waitUntilTrue(() => {
        try {
          val data = mirrorMakerConsumer.receive()
          data.topic == topic && new String(data.value) == msg
        } catch {
          case _: ConsumerTimeoutException => false// swallow it
        }
      }, "MirrorMaker consumer should be able to subscribe the correct topic and read the correct message.")
    } finally {
      consumer.close()
    }`
Is that okay to you then? Thanks.